### ISSUE TITLE
later library consumes more cpu?

### ISSUE THREAD

**Issue by marwellit** [hzeller/rpi-rgb-led-matrix#517]
hi

I'm in the process of building 4 identical 4 panel signs

I've did 3 before Christmas - all share same Raspbian Stretch Lite version, python script , hardware and library version  and all 3 run the CPU at 48% - the LED library used was from (I think) November sometime.....

Ive just done the 4th panel and thought I'd update the library to the current one but I observe that the cpu usage is now 75%.......everything else is unchanged...if I copy the library over from one of the other older panels the cpu drops back to 50%

any ideas why this seems to be the case?

thanks
Dave

**Comment by hzeller** [comment]
Yes - recently I've increased the fraction the library does busy waiting on a timing context ( https://github.com/hzeller/rpi-rgb-led-matrix/commit/181e3bf3a2d1c185edf18c31458afb3e2217a73a ), so this increased the CPU you see.

However, this should not influence any performance of anything you do, because this _only_ affects the CPU-core the LED matrix update is happening (core 3). So if you have set the `isolcpus=3` setting as described in the README, this is the only thing this core is running anyway.

Bottom line: nothing to worry about, just a measure to slightly improve brightness fluctuations.
